contract/src/test/kotlin/io/provenance/scope/loan/contracts/RecordLoanContractUnitTest.kt

```diff
@@ -0,0 +1,165 @@
+package io.provenance.scope.loan.contracts
+
```

Comment:
Nice job being pragmatic and creating the unit tests, implementing some logic and marking some as TODO.